processing/src/main/java/org/apache/druid/frame/write/FrameWriterUtils.java

```diff
@@ -229,7 +229,11 @@ public static void copyByteBufferToMemory(
       final byte b = src.get(p);
 
       if (!allowNullBytes && b == 0) {
-        throw new InvalidNullByteException();
+        throw new InvalidNullByteException(
+            "Unable to add the frame because it contains null bytes. This usually happens when the added string columns "
```

Comment:
Thanks for the improved message. The user, however, knows nothing about frames. Can we word this from the user's perspective?

`Druid does not support null (0x00) bytes in strings. File %s, row $d, column %s contains null bytes: [%s].`

The string would be encoded so that control characters appear as `\U0000` so the user can see the position of the null bytes.

Maybe we don't know the row number (in a form useful to the user.) If not, just list the column.

This is a case of unparsable data. Should we have caught it at the time we _read_ the data rather than when _writing_ to a frame? Should we invoke our bad-row logic to skip this row? That logic should log the bad row for later re-ingestion, but I don't think we've added that ability.

If we catch the problem on read, then the check here is more of an assertion. Though, perhaps the data was created by an expression, so it is still worth validating.